### PR TITLE
Adhering to OSB API changes :Feature/osbapi

### PR DESCRIPTION
**Pull request by sauravmndl**
Created PR #131 earlier and reverted in #188 
Fixes issue #193

### CODE REVIEW

**Comment by sauravmndl** [comment]
Please 'Create a merge commit' not 'Squash and merge'.
Let me know at the time of merging.

File: lib/controllers/ServiceFabrikApiController.js [hunk 1]
```diff
@@ -457,10 +472,11 @@ class ServiceFabrikApiController extends BaseController {
       .omit('repeatInterval')
       .set('instance_id', req.params.instance_id)
       .set('trigger', CONST.BACKUP.TRIGGER.SCHEDULED)
-      .set('space_guid', req.entity.space_guid)
+      .set('tenant_id', req.entity.tenant_id)
+      .set('plan_id', req.manager.plan.id)
       .set('service_id', req.manager.service.id)
       .value();
-    return this.cloudController.getOrgAndSpaceDetails(data.instance_id, data.space_guid)
+    return this.cloudController.getOrgAndSpaceDetails(data.instance_id, data.tenant_id) // Why all these needed ?
```
Comment (subhankarc) [inline lib/controllers/ServiceFabrikApiController.js]: Remove the comment if it's clarified?

File: lib/controllers/DashboardController.js [hunk 2]
```diff
@@ -121,8 +121,10 @@ class DashboardController extends BaseController {
     const service_id = req.params.service_id;
     const plan_id = req.params.plan_id;
     logger.info(`Validating service '${service_id}' and plan '${plan_id}'`);
+    const operation = utils.decodeBase64(_.get(req, 'query.operation') || 'bnVsbA==');
```
Comment (kamath-prasad) [inline lib/controllers/DashboardController.js]: please move `bnVsbA==` to constants as it is difficult to understand what this string means.

File: lib/controllers/ServiceBrokerApiController.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
 const ServiceBindingAlreadyExists = errors.ServiceBindingAlreadyExists;
 const ServiceBindingNotFound = errors.ServiceBindingNotFound;
 const ContinueWithNext = errors.ContinueWithNext;
+const CONST = require('../constants');
 
 class ServiceBrokerApiController extends BaseController {
```
Comment (subhankarc) [inline lib/controllers/ServiceBrokerApiController.js]: Can you also change the minimum broker API version? CC now sends 2.13 so it is good to update it.

File: lib/controllers/ServiceFabrikApiController.js (hunk 1 above)
Comment (kamath-prasad) [inline lib/controllers/ServiceFabrikApiController.js]: can you remove the comment `Why all these needed ?`

Comment (sauravmndl) [reply lib/controllers/ServiceFabrikApiController.js]: done

File: lib/controllers/ServiceBrokerApiController.js (hunk 3 above)
Comment (sauravmndl) [reply lib/controllers/ServiceBrokerApiController.js]: Done

File: lib/controllers/DashboardController.js (hunk 2 above)
Comment (sauravmndl) [reply lib/controllers/DashboardController.js]: done

File: lib/controllers/ServiceFabrikApiController.js (hunk 1 above)
Comment (sauravmndl) [reply lib/controllers/ServiceFabrikApiController.js]: done

File: lib/controllers/ServiceBrokerApiController.js [hunk 4]
```diff
@@ -23,8 +24,8 @@ class ServiceBrokerApiController extends BaseController {
 
   apiVersion(req, res) {
     /* jshint unused:false */
-    const minVersion = '2.9';
-    const maybeVersion = '2.8';
+    const minVersion = CONST.SF_BROKER_API_VERSION_MIN;
+    const maybeVersion = CONST.SF_BROKER_API_VERSION_MAYBE;
     const minCloudControllerApiVersion = '2.57.0';
     const version = _.get(req.headers, 'x-broker-api-version', '1.0');
     return Promise
```
Comment (jagadish-kb) [inline lib/controllers/ServiceBrokerApiController.js]: This method checks for CF version. This method still has to be refactored with PlatformManager. . For now can conditionally allow for this to be approved. But please add this as a pending Item in Cross-Consumption story.

Comment (sauravmndl) [reply lib/controllers/ServiceBrokerApiController.js]: Created issue #194 . Will update the issue.

**Pull request merged**